tests/test_api/test_worker.py

```diff
@@ -211,6 +220,65 @@ def testApiWorker_NamespaceReplacement(filesDir):
 
 # END Test testApiWorker_NamespaceReplacement
 
+
+@pytest.mark.api
+def testApiWorker_NamespaceRejectedIfWrongEnvStaging(filesDir):
+    """Test rejection if namespace contains a .staging not matching the env."""
+
+    xsdFile = os.path.join(filesDir, "mmd", "mmd.xsd")
+    passFile = os.path.join(filesDir, "api", "staging.xml")
+
+    xsdObj = lxml.etree.XMLSchema(lxml.etree.parse(xsdFile))
+    passWorker = Worker("none", passFile, xsdObj)
+
+    passWorker._conf.env_string = ""
+
+    # Valid XML
+    passData = bytes(readFile(passFile), "utf-8")
+    valid, msg, passData = passWorker.validate(passData)
+    assert valid is False
+    assert msg == "Namespace test.no.staging does not match the env "
+
+    passWorker._conf.env_string = "staging"
+
+    # Valid XML
+    passData = bytes(readFile(passFile), "utf-8")
+    valid, msg, passData = passWorker.validate(passData)
+    assert valid is True
+
+
+# END Test testApiWorker_NamespaceRejectedIfWrongEnvStaging
+
+
+@pytest.mark.api
```

Comment:
So, the test for staging and dev is basically the same, right? The only diff is in env name? You could do that in the same test to avoid all the code duplication. And add a check that "no.met" or similar is accepted in both.

We should also check that the test files are not accepted in prod.

Reply:
they are getting a different passFile,   
the fact that they are not accepted in prod is already the point of the tests